### ISSUE TITLE
Ubuntu : .NET Core 3.1 build error

### ISSUE THREAD

**Issue by deokershesh** [dotnet/corefx#42802]
I downloaded .NET Core 3.1 source code, while doing the build.sh on Ubuntu, it reports following errors! There doesn't seem to be good error message to possibly resolve the same. Kindly help to fix the below build issue.

root@64ac36d6c74e:/shesh/dotnetcore/corefx-release-3.1# uname -a
Linux 64ac36d6c74e 4.9.184-linuxkit #1 SMP Tue Jul 2 22:58:16 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

  Setting up directories for build
  Checking pre-requisites...
  Commencing build of corefx native components for Linux.x64.Debug
  Invoking "/shesh/dotnetcore/corefx-release-3.1/src/Native/Unix/gen-buildsys-clang.sh" "/shesh/dotnetcore/corefx-release-3.1/src/Native/Unix" "3" "9" "x64" "DEBUG" " -DFEATURE_DISTRO_AGNOSTIC_SSL=1 -DCMAKE_STATIC_LIB_LINK=0"
  -- Configuring incomplete, errors occurred!
  See also "/shesh/dotnetcore/corefx-release-3.1/artifacts/obj/native/netcoreapp-Linux-Debug-x64/CMakeFiles/CMakeOutput.log".
  See also "/shesh/dotnetcore/corefx-release-3.1/artifacts/obj/native/netcoreapp-Linux-Debug-x64/CMakeFiles/CMakeError.log".
  Failed to generate native component build project!
/shesh/dotnetcore/corefx-release-3.1/src/Native/build-native.proj(36,5): error MSB3073: The command ""/shesh/dotnetcore/corefx-release-3.1/src/Native/build-native.sh" x64 Debug Linux outconfig netcoreapp-Linux-Debug-x64 -portable" exited with code 1.

Build FAILED.

/shesh/dotnetcore/corefx-release-3.1/src/Native/build-native.proj(36,5): error MSB3073: The command ""/shesh/dotnetcore/corefx-release-3.1/src/Native/build-native.sh" x64 Debug Linux outconfig netcoreapp-Linux-Debug-x64 -portable" exited with code 1.
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:46.11
Build failed (exit code '1').
root@64ac36d6c74e:/shesh/dotnetcore/corefx-release-3.1#

**Comment by danmosemsft** [comment]
@deokershesh do you need to build 3.1? If you want latest sources, please clone https://github.com/dotnet/runtime instead, where the sources have moved. See
https://github.com/dotnet/runtime/blob/master/docs/workflow/README.md

**Comment by deokershesh** [comment]
@danmosemsft, Infact I started with the same, please refer https://github.com/dotnet/runtime/issues/1207 , however I was informed that, its for future .Net Core 5.0 project and if I am looking for .NET Core 3.1 then I should be using  https://github.com/dotnet/corefx/tree/release/3.1, which now I am using. I have built .NET Core 3.1 successfully on Windows but Linux/Ubuntu!!

Any help in this regard (Ubuntu build error) would be highly appreciated. Basically, I am working to support Informix database by building .NET Core Provider (replica of System.Data.Odbc customized for Informix).

**Comment by danmosemsft** [comment]
@deokershesh got it. Ok have you checked you have the prerequisites? Try upgrading your relevant packages first.

https://github.com/dotnet/corefx/blob/release/3.1/Documentation/building/unix-instructions.md#user-content-linux

**Comment by deokershesh** [comment]
Thanks @danmosemsft, The above link helped and I tried to refresh/install the suggested packages and it worked! Thanks a lot. Now, I need to customize for Informix, hope it will also go smooth like on Windows!

**Comment by danmosemsft** [comment]
@deokershesh sure. Post back how your project goes. Do you plan to share it for others to take advantage of?

**Comment by deokershesh** [comment]
Yes @danmosemsft, We will host on Nuget. I am able to build Informix .NET Core Provider on Linux as well by the way. Facing some run time issue, yet to investigate in detail. By the way, what's the best tool to debug C#(.cs) files on Linux?

**Comment by danmosemsft** [comment]
VS Code. This may help although it might be out of date-
https://github.com/dotnet/corefx/blob/release/3.1/Documentation/debugging/unix-instructions.md

**Comment by deokershesh** [comment]
Actually I am using "docker image" (/bin/bash command prompt).

**Comment by wfurt** [comment]
What is in CMakeOutput.log @deokershesh? Also  #42812 suggests you solved your build problem. Should this issue be closed?
